### PR TITLE
[InternalComponent] Add is_link_mode to internal component output

### PR DESCRIPTION
**Pull request by lalala123123**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py [hunk 1]
```diff
@@ -106,10 +106,18 @@ def _from_base(cls, _input: Union[Input, Dict]) -> Optional["InternalInput"]:
...
-    def __init__(self, *, datastore_mode=None, **kwargs):
+    def __init__(self, *, datastore_mode=None, is_link_mode=None, **kwargs):
         self.datastore_mode = datastore_mode
+        self.is_link_mode = is_link_mode
         super().__init__(**kwargs)
 
+    def _to_dict(self, remove_name=True):
+        """Convert the Output object to a dict."""
+        result = super()._to_dict(remove_name)
+        if self.is_link_mode is not None:
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py:117]: why datastore_mode don't need this?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py [hunk 2]
```diff
@@ -106,10 +106,18 @@ def _from_base(cls, _input: Union[Input, Dict]) -> Optional["InternalInput"]:
...
+            result["is_link_mode"] = self.is_link_mode
+        return result
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py:119]: why input does not have this logic?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py [hunk 3]
```diff
@@ -106,10 +106,18 @@ def _from_base(cls, _input: Union[Input, Dict]) -> Optional["InternalInput"]:
 
 
 class InternalOutput(Output):
-    def __init__(self, *, datastore_mode=None, **kwargs):
+    def __init__(self, *, datastore_mode=None, is_link_mode=None, **kwargs):
         self.datastore_mode = datastore_mode
+        self.is_link_mode = is_link_mode
         super().__init__(**kwargs)
 
+    def _to_dict(self, remove_name=True):
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py:114]: it's not recommended to add parameters to _to_dict. This is a function inherited from a mixin and adding params may break the inheritance.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py [hunk 4]
```diff
@@ -106,10 +106,18 @@ def _from_base(cls, _input: Union[Input, Dict]) -> Optional["InternalInput"]:
 
 
 class InternalOutput(Output):
-    def __init__(self, *, datastore_mode=None, **kwargs):
+    def __init__(self, *, datastore_mode=None, is_link_mode=None, **kwargs):
         self.datastore_mode = datastore_mode
+        self.is_link_mode = is_link_mode
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py:111]: has it been added to corresponding Schema?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py (hunk 3 above)
Comment (elliotzh) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py:114]: so what is the remove name for?

Comment (elliotzh) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py:114]: usually you won't need to override this function if schema & related configurations are correctly set

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py (hunk 4 above)
Comment (lalala123123) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_input_outputs.py:111]: It has been added to the InternalOutputSchema

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**